File: JavaScript/455-Assign Cookies.js
/**
 * @param {number[]} g
 * @param {number[]} s
 * @return {number}
 */
var findContentChildren = function(g, s) {
    let index_g = 0, index_s = 0;
    
    g.sort(function(a,b){return a-b;})
    s.sort(function(a,b){return a-b;})
    
    while (index_g < g.length && index_s < s.length){
        if (s[index_s] < g[index_g]){
            index_s++;
        }else{
            index_s++;
            index_g++;
        }
    }
    return index_g
    
};

File: C++/455-Assign Cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int idx1 = 0, idx2 = 0;
        
        while (idx1 < g.size() && idx2 < s.size()){
            if (s[idx2] >= g[idx1]){
                idx1++;
            }
            idx2++;
        }
        return idx1;
    }
};